src/zenml/client.py

```diff
@@ -2510,30 +2442,69 @@ def list_pipelines(
         )
 
     def get_pipeline(
-        self, name_id_or_prefix: Union[str, UUID]
+        self,
+        id: Optional[UUID] = None,
+        name: Optional[str] = None,
+        version: Optional[str] = None,
     ) -> PipelineResponseModel:
         """Get a pipeline by name, id or prefix.
 
         Args:
-            name_id_or_prefix: The name, id or prefix of the pipeline.
+            id: The id of the pipeline.
+            name: The name of the pipeline.
+            version: The pipeline version. If left empty, will return
+                the latest version.
 
         Returns:
             The pipeline.
+
+        Raises:
+            ValueError: If both name and ID are unset.
+            KeyError: If no pipelines were found for the given name and version.
         """
-        return self._get_entity_by_id_or_name_or_prefix(
-            get_method=self.zen_store.get_pipeline,
-            list_method=self.list_pipelines,
-            name_id_or_prefix=name_id_or_prefix,
-        )
+        if id and (name or version):
+            logger.warning(
```

Comment:
I think it's just wrong usage of the method if you supply both name and ID, you might expect that the returned pipeline has the given name but it could be anything depending on the ID